cmd/dataobj-inspect/dump.go

```diff
@@ -0,0 +1,126 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/alecthomas/kingpin/v2"
+	"github.com/fatih/color"
+	"github.com/prometheus/prometheus/model/labels"
+
+	"github.com/grafana/loki/v3/pkg/dataobj"
+	"github.com/grafana/loki/v3/pkg/dataobj/sections/logs"
+	"github.com/grafana/loki/v3/pkg/dataobj/sections/streams"
+)
+
+// dumpCommand dumps the contents of the data object.
+type dumpCommand struct {
+	files *[]string
+}
+
+func (cmd *dumpCommand) run(c *kingpin.ParseContext) error {
+	for _, f := range *cmd.files {
+		cmd.dumpFile(f)
+	}
+	return nil
+}
+
+func (cmd *dumpCommand) dumpFile(name string) {
+	f, err := os.Open(name)
+	if err != nil {
+		exitWithError(fmt.Errorf("failed to open file: %w", err))
+	}
+	defer func() { _ = f.Close() }()
+	fi, err := f.Stat()
+	if err != nil {
+		exitWithError(fmt.Errorf("failed to read fileinfo: %w", err))
+	}
+	dataObj, err := dataobj.FromReaderAt(f, fi.Size())
+	if err != nil {
+		exitWithError(fmt.Errorf("failed to read dataobj: %w", err))
+	}
+	for offset, sec := range dataObj.Sections() {
+		switch {
+		case streams.CheckSection(sec):
+			cmd.dumpStreamsSection(context.TODO(), offset, sec)
+		case logs.CheckSection(sec):
+			cmd.dumpLogsSection(context.TODO(), offset, sec)
+		default:
+			exitWithError(errors.New("unknown section"))
```

Comment:
Do we have the section metadata from sec? It might be nice to print "unknown section: X" instead of exiting completely on the first unknown section